### DISCUSSION TITLE
Exclude everything in a directory except a subdir and its files

### DISCUSSION THREAD

**Discussion by rieje** [sharkdp/fd#1628] [Q&A]
I'm trying to exclude all files in `~/.firejail` but include all files in `~/.firejail/<any dir>/Downloads/` . The following in `.ignore` doesn't seem to do that and ignores all files in `~/.firejail`, any ideas?

    /.firejail/ 
    !/.firejail/**/Downloads/*

If I uncomment the first one `/.firejail/` then all its files show. Looking at the .gitignore syntax this is what I understand should work, i.e. "exclude everything in `/.firejail/`, except those in `/.firejail/**/Downloads/*`.

P.S. If I pass in multiple ignore files and some of them share the same rules, is that a problem and I need to resolve them (e.g. read all the rules, removing duplicates then write them out to a tmp file and pass that to `--ignore-file`?

**Comment by tmccombs** [comment]
The way gitignore, and fd ignore files, works is if a folder is excluded, it isn't traversed at all. So there isn't a way to re-add a child of an ignored file.

Maybe you could do something like:

```
/.firejail/**
!/.firejail/**/Downloads/*
```

Although then .firejail itself would be included

**Reply by rieje** [reply]
That doesn't show any results under /.firejail except /.firejail itself. Upon looking around further, it doesn't seem to be possible with gitignore syntax. My workaround is to just ignore the parent (/.firejail), then give fd the arguments `~/.firejail/**/Downloads/` (techinically there's a limit to CLI arguments so it's not scalable).
